logicalRequirements.md

```diff
@@ -343,6 +343,68 @@ __Additional considerations__
+
+#### comeInWithGMode object
+A `comeInWithGMode` object represents the need to either have or obtain G-mode when entering the room. It has the following properties:
+* _fromNodes:_ Indicates from what doors this logical requirement expects Samus to enter the room.
+* _mode:_ Takes one of three possible values, "direct", "indirect", or "any", indicating whether this logical requirement expects Samus to enter in direct G-mode, indirect G-mode, or either. Direct G-mode is the state obtained when G-mode is first entered (i.e., the next room after the G-mode setup is performed), while indirect G-mode is the state after passing a door transition with G-mode (usually back into the room where the G-mode setup was performed).
+* _artificialMorph:_ A boolean indicating whether the logical requirement expects Samus to either obtain or already have an artificially morphed state when coming into the room, or to have collected the Morph item.
+
+__Example:__
+```json
+{"comeInWithGMode": {
+  "fromNodes": [1],
+  "mode": "any",
+  "artificialMorph": false
+}}
+```
+
+__Additional considerations__
+
+* A `comeInWithGMode` object implicitly requires X-Ray Scope and a Reserve Tank.
+* A `comeInWithGMode` object implicitly requires the `canEnterGMode` tech.
+  * If `artificialMorph` is `true` then it also requires the `canArtificialMorph` tech.
+* A `comeInWithGMode` requires that one of the indicating nodes in `fromNodes` has a matching `leaveWithGModeSetup` or `leaveWithGMode` object in the corresponding door node of the neighboring room:
+  * A `leaveWithGModeSetup` object must satisfy following requirements in order to match:
+    * The `mode` in the `comeInWithGMode` object must be "direct" or "any".
+    * Samus must have non-zero reserve energy.
+    * Any additional requirements in the `requires` property of the `leaveWithGModeSetup`.
+  * A `leaveWithGMode` object must satisfy the following requirements in order to match:
+    * The `mode` in the `comeInWithGMode` object must be "indirect" or "any".
+    * If `artificialMorph` is `true`, then either the `leavesWithArtificialMorph` property of the `leaveWithGMode` object must be `true` or there is an additional requirement that the Morph item be collected.
+    * Any additional requirements in the `requires` property of the `leaveWithGMode` object.
+* In the case of direct G-mode, `comeInWithGMode` object implicitly requires an energy drain caused by the reserve trigger and preceding setup:
+  * If the tech `canEnterGModeImmobile` is enabled and the `gModeImmobile` on the corresponding door is satisfied, then Samus' regular energy will become whatever reserve energy she had before the transition, truncated to her maximum amount of regular energy (based on the number of ETanks collected).
+  * Otherwise, Samus' regular energy will become 4, or whatever reserve energy she had before the transition if it was less than 4.
```

Comment:
Hmm, how is it backwards? The way I'm thinking of it, before executing the strat to enter direct G-mode, reserve energy can be any positive value (doesn't have to be between 1 and 4), because it is always possible to drain it down into the correct range (4 or less) as part of the strat.

Reply:
The way I see it, the `canEnterGModeImmobile` description would apply where reserve->regular energy.  And the main difference is that reserve energy is 1to4 at the point of activating GMode, which could use some stronger wording.  

Manually draining energy would fit under `leaveWithGModeSetup` tbh.
I don't think `reflecting the need` improves the description.

But its not at all an important difference.

Reply:
Ah, yeah "reflecting the need" was me starting to type something and then getting sidetracked, haha. Thanks. I added a little more above now to clarify how the reduced energy is coming from the setup in the previous room.